app/models/aggregation.rb

```diff
@@ -14,8 +14,11 @@ class Aggregation < ActiveRecord::Base
   validate :aggregation, :workflow_version_present
 
   def self.scope_for(action, user, opts={})
-    return super unless (action == :show || action == :index) && !user.is_admin?
-    joins(:workflow).merge(Workflow.scope_for(:update, user, opts))
+    if (action == :show || action == :index) && !user.is_admin?
+      joins(:workflow).merge(Workflow.scope_for(:update, user, opts))
```

Comment:
I think you can make your controller much lighter by doing

```ruby
joins(:workflow).merge(Workflow.scope_for(:update, user, opts)).or('"workflow"."aggregation"->>'public' = 'true')
```

But you'd have to index the aggregation json blob.